pkg/minishift/openshift/version/openshift_versions.go

```diff
@@ -90,28 +76,35 @@ func PrintDownStreamVersions(output io.Writer, minSupportedVersion string) {
 	}
 }
 
-func PrintUpStreamVersions(output io.Writer, minSupportedVersion string) {
-	resp, err := getResponseBody("https://registry.hub.docker.com/v2/repositories/openshift/origin/tags/")
+func PrintUpStreamVersions(output io.Writer, minSupportedVersion string, defaultVersion string) {
+	dockerRegistryUrl := "https://registry.hub.docker.com/v1/repositories/openshift/origin/tags"
+	resp, err := getResponseBody(dockerRegistryUrl)
 	if err != nil {
 		fmt.Fprintf(output, "Error Occured %s", err)
 		return
 	}
 	defer resp.Body.Close()
 	decoder := json.NewDecoder(resp.Body)
-	var data ImageTags
+	var data []ImageTags
 	err = decoder.Decode(&data)
 	if err != nil {
 		fmt.Printf("%T\n%s\n%#v\n", err, err, err)
 		return
 	}
 	fmt.Fprint(output, "The following OpenShift versions are available: \n")
 	var tagsList []string
-	for _, imageinfo := range data.Results {
+	for _, imageinfo := range data {
 		if strings.Contains(imageinfo.Name, "latest") {
 			continue
 		}
 		if util.VersionOrdinal(imageinfo.Name) >= util.VersionOrdinal(minSupportedVersion) {
-			tagsList = append(tagsList, imageinfo.Name)
+			if util.VersionOrdinal(imageinfo.Name) > util.VersionOrdinal(defaultVersion) {
```

Comment:
@LalatenduMohanty We should use semver logic for upstream which is part of clusterup code https://github.com/minishift/minishift/blob/master/pkg/minishift/clusterup/clusterup.go#L282 `VersionOrdinal` only for downstream because we don't follow there semver logic.